src/module/item/condition/document.ts

```diff
@@ -241,20 +241,21 @@ class ConditionPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends
         const conditions = this.actor.conditions.active;
 
         const ofSameType = conditions.filter((c) => c !== this && c.key === this.key);
-        for (const condition of ofSameType) {
-            if (condition.slug === "persistent-damage") {
+        for (const other of ofSameType) {
+            const otherIsGTE = other.value === this.value || (other.value && this.value && other.value >= this.value);
+            if (other.slug === "persistent-damage") {
                 const thisValue = this.system.persistent?.expectedValue ?? 0;
-                const otherValue = condition.system.persistent?.expectedValue ?? 0;
-                if (thisValue >= otherValue) deactivate(condition);
-            } else if (this.value === condition.value && this.inMemoryOnly) {
-                // Defer to other, identical condition if this one is in-memory-only
+                const otherValue = other.system.persistent?.expectedValue ?? 0;
+                if (thisValue >= otherValue) deactivate(other);
+            } else if (otherIsGTE && this.inMemoryOnly) {
+                // Defer to other, gte condition if this one is in-memory-only
                 deactivate(this);
-            } else if (this.value === condition.value && (!this.isLocked || condition.isLocked)) {
+            } else if (otherIsGTE && (!this.isLocked || other.isLocked)) {
```

Comment:
This logic can lead to other getting deactivated if other is greater. Perhaps flipping the logic such that you check `!other.isLocked || this.isLocked` instead and then `deactivate(this)` might suffice?

Personally, I think it might be best to do a batch operation (group by then process each group) in the actor if you can swing the extra time. Trying to dedupe data on an item by item basis is pretty hard to reason around.